src/components/directorybrowser/directorybrowser.js

```diff
@@ -1,51 +1,148 @@
         var html = "";
-        return html += '<div class="listItem listItem-border ' + cssClass + '" data-type="' + type + '" data-path="' + path + '">', html += '<div class="listItemBody" style="padding-left:0;padding-top:.5em;padding-bottom:.5em;">', html += '<div class="listItemBodyText">', html += name, html += "</div>", html += "</div>", html += '<i class="md-icon" style="font-size:inherit;">arrow_forward</i>', html += "</div>"
+        html += '<div class="listItem listItem-border ' + cssClass + '" data-type="' + type + '" data-path="' + path + '">';
+        html += '<div class="listItemBody" style="padding-left:0;padding-top:.5em;padding-bottom:.5em;">';
+        html += '<div class="listItemBodyText">';
+        html += name;
+        html += "</div>";
+        html += "</div>";
+        html += '<i class="md-icon" style="font-size:inherit;">arrow_forward</i>';
+        html += "</div>";
+        return html;
     }
 
     function getEditorHtml(options, systemInfo) {
         var html = "";
-        if (html += '<div class="formDialogContent scrollY">', html += '<div class="dialogContentInner dialog-content-centered" style="padding-top:2em;">', !options.pathReadOnly) {
+        html += '<div class="formDialogContent scrollY">';
+        html += '<div class="dialogContentInner dialog-content-centered" style="padding-top:2em;">';
+        if (!options.pathReadOnly) {
             var instruction = options.instruction ? options.instruction + "<br/><br/>" : "";
-            html += '<div class="infoBanner" style="margin-bottom:1.5em;">', html += instruction, html += Globalize.translate("MessageDirectoryPickerInstruction").replace("{0}", "<b>\\\\server</b>").replace("{1}", "<b>\\\\192.168.1.101</b>"), "synology" === (systemInfo.PackageName || "").toLowerCase() ? (html += "<br/>", html += "<br/>", html += '<a is="emby-linkbutton" class="button-link" href="https://web.archive.org/web/20181216120305/https://github.com/MediaBrowser/Wiki/wiki/Synology-:-Setting-Up-Your-Media-Library-Share" target="_blank">' + Globalize.translate("LearnHowToCreateSynologyShares") + "</a>") : "bsd" === systemInfo.OperatingSystem.toLowerCase() ? (html += "<br/>", html += "<br/>", html += Globalize.translate("MessageDirectoryPickerBSDInstruction"), html += "<br/>", html += '<a is="emby-linkbutton" class="button-link" href="http://doc.freenas.org/9.3/freenas_jails.html#add-storage" target="_blank">' + Globalize.translate("ButtonMoreInformation") + "</a>") : "linux" === systemInfo.OperatingSystem.toLowerCase() && (html += "<br/>", html += "<br/>", html += Globalize.translate("MessageDirectoryPickerLinuxInstruction"), html += "<br/>"), html += "</div>"
+            html += '<div class="infoBanner" style="margin-bottom:1.5em;">';
+            html += instruction;
+            html += Globalize.translate("MessageDirectoryPickerInstruction").replace("{0}", "<b>\\\\server</b>").replace("{1}", "<b>\\\\192.168.1.101</b>");
```

Comment:
I wonder what these `.replace()` calls do...

Reply:
Well the message is 
`Network paths can be entered manually in the event the Network button fails to locate your devices. For example, {0} or {1}.`
and each backslash is escaped, so this just turns into
`Network paths can be entered manually in the event the Network button fails to locate your devices. For example, \\server or \\192.168.1.101.`
I guess they just wanted to be able to update the examples without changing the .json files for all the languages.

Reply:
That looks ugly, but that's not your fault, so let's leave it be for now...